avm/res/operational-insights/workspace/main.bicep

```diff
@@ -496,4 +470,22 @@ type diagnosticSettingType = {
 
   @description('Optional. The full ARM resource ID of the Marketplace resource to which you would like to send Diagnostic Logs.')
   marketplacePartnerResourceId: string?
-}[]?
+}
+
+@export()
+type gallerySolutionType = {
```

Comment:
I wonder if we should implement this type in the `solution` module as an exported type and import it all over. Thoughts?

Reply:
@AlexanderSehr thanks for the review! This sounds reasonable. Do you suggest to re-design the parameters of the `solution` module to use this type, or just to create an exportable type for cross-referencing purposes without changing the rest of the module?

Reply:
I created #3718, once it's merged, I will update this PR

Reply:
Hey @krbar,
I'd suggest what you did - with the one caveat that I commented in that PR. Thanks :)

Reply:
@AlexanderSehr I've updated the module to use the newest `solution` module and the type definition from there. Please have a look.